src/main.cpp

```diff
@@ -4548,14 +4798,31 @@ bool ProcessNewBlock(CValidationState& state, CNode* pfrom, CBlock* pblock, CDis
         }
 
         // Store to disk
-        CBlockIndex* pindex = NULL;
+        CBlockIndex* pindex = nullptr;
         bool ret = AcceptBlock (*pblock, state, &pindex, dbp, checked);
         if (pindex && pfrom) {
             mapBlockSource[pindex->GetBlockHash ()] = pfrom->GetId ();
         }
         CheckBlockIndex ();
-        if (!ret)
-            return error ("%s : AcceptBlock FAILED", __func__);
+        if (!ret) {
```

Comment:
Spam check only occurs if it's an invalid block, if the block is valid and we already have it, it could be due our sync process that use to request the same block multiple times, better to not ban good nodes, at least for now.

When the sync process is totally polished then we could change this for sure.